pkg/internalapis/cnsvolumeoperationrequest/cnsvolumeoperationrequest.go

```diff
@@ -120,22 +107,124 @@ func InitVolumeOperationRequestInterface(ctx context.Context) (VolumeOperationRe
 // Returns an error if any error is encountered while attempting to
 // read the previously persisted information from the API server.
 // If no previous information has been persisted, it returns an empty object.
-func (or *operationRequestStoreOnETCD) GetRequestDetails(ctx context.Context, name string) (*VolumeOperationRequestDetails, error) {
+func (or *operationRequestStore) GetRequestDetails(ctx context.Context, name string) (*VolumeOperationRequestDetails, error) {
 	log := logger.GetLogger(ctx)
-	// TODO: Implement GetRequestDetails for operationRequestStoreOnETCD
-	err := fmt.Errorf("GetRequestDetails not implemented for operationRequestStoreOnETCD")
-	log.Error(err)
-	return nil, err
+	namespacedName := client.ObjectKey{Name: name, Namespace: csiconfig.DefaultCSINamespace}
+	log.Infof("Getting CnsVolumeOperationRequest instance with name %s/%s", namespacedName.Namespace, namespacedName.Name)
+
+	instance := &cnsvolumeoperationrequestv1alpha1.CnsVolumeOperationRequest{}
+	err := or.k8sclient.Get(ctx, namespacedName, instance)
+	if err != nil {
+		// If instance does not exist, return an empty object.
+		if apierrors.IsNotFound(err) {
+			log.Infof("CnsVolumeOperationRequest instance %s/%s not found. Returning ...", namespacedName.Namespace, namespacedName.Name)
+			return nil, nil
+		}
+		log.Errorf("failed to get CnsVolumeOperationRequest instance %s/%s with error: %v", namespacedName.Namespace, namespacedName.Name, err)
+		return nil, err
+	}
+	log.Debugf("Found CnsVolumeOperationRequest instance %v", spew.Sdump(instance))
+
+	// Determine which operation details need to be returned. Callers only need to know about the
+	// last operation that was invoked on a volume.
+	var operationDetailsToReturn cnsvolumeoperationrequestv1alpha1.OperationDetails
+	if len(instance.Status.LatestOperationDetails) != 0 {
+		operationDetailsToReturn = instance.Status.LatestOperationDetails[len(instance.Status.LatestOperationDetails)-1]
```

Comment:
I think we need a lock to ensure we retrieve the latest LatestOperationDetails.

Reply:
`instance` is a local copy of this object so i dont think we need a lock